docs/Contribution.md

```diff
@@ -0,0 +1,62 @@
+
+## Components should be functional
+
+As detailed in the [architecture documentation](./Architecture.md), all react components should be functional components whos model logic is provided purely through Hooks. This allows us to follow the MVC pattern and gives us the power to use swappable view layers as all views are responsible for **only** rendering.
+
+## Use Contexts for state management
+
+Any data that needs to be shared across components that can exist as state should be put into a context. Contexts allow data to be provided at a higher level and then consumed by any child component that wants access to it. This is a much cleaner solution to prop-drilling and other libraries such as react-redux. [View the context documentation here](../client/Contexts/README.md).
+
+## All components should have a Storybook
+
+Every component should also come with a detailed Storybook. They should have appropriate knobs and actions that show the full capability of the component. Developing components with Storybook is a great way to make sure most of this gets done as part of the process, instead of being an afterthough. For more details on Storybook [view our Storybook documentation](./Architecture.md#storybook).
+
+## Component Conventions
+
+We have defined four main types of components that make up the UI: Elements, Groups, Panels, and Bootstrap. These defnitions allow us to clearly split up complexity and allow for maximum modularity of the UI. For full information about why we use this structure [view our architecture documentation](./Architecture.md#element-group-panel-bootstrap-component-pattern)
+
+## CSS should be written using the block-element-modifier (BEM) naming convention
+
+All CSS in this project is written using the BEM naming convetion. We chose BEM for a variety of reasons
+
+- It is a widely used convention that should be quick to pick up my most experienced UI developers.
+- It makes JSX much more readable when the styling applied can be derived from the class names.
+- It should help reduce duplicate CSS as common styles will already be defined.
+
+BEM is dificult to enforce automatically so reviewers should be vigilant to CSS class names to ensure it matches the BEM convention.
+
+## Component CSS should live in its own namespace
+
+As this project uses SCSS as a CSS precompiler, it makes namespacing CSS easy. Each component should silo off its own CSS by encapsulating the file in a single class that matches the name of the component. For example, the JSX would look like this:
+
+```
+<div className={'myComponent'}>
+  <button className={'button-primary'}>I am a child element</button>
+  // other children
+</div>
+```
+
+And the corresponding SCSS should look like this:
+
+```
+.myComponent{
+  .button-primary {
+
+  }
+  // other child css
+}
+// EOF
+```
```

Comment:
Do we want to Have the CSS in a coding doc that's being added in PR #29 once it's merged?  @matthew-chirgwin @JQrdan

Reply:
I think so - the tooling added will help enforce, but examples are always good too